container/openejb-loader/src/main/java/org/apache/openejb/loader/FileUtils.java

```diff
@@ -20,18 +20,25 @@
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.util.Hashtable;
+import java.util.concurrent.ThreadLocalRandom;
 
 @SuppressWarnings("PMD.OverrideBothEqualsAndHashcode")
 public class FileUtils {
 
-    private static final java.util.Random _random = new java.util.Random();
+    private static final ThreadLocalRandom _random = ThreadLocalRandom.current();
```

Comment:
About the name, sorry my mistake, that is a const, therefore `RANDOM` sounds nice

Reply:
:)

Reply:
Not this isn't safe usage of `ThreadLocalRandom`.  Whatever thread initializes `FileUtils` will have its not-thread-safe instance of ThreadLocalRandom saved in a static field and shared thereafter by all threads.  In short, this code grabs what is intended to be thread-scoped and stores it into the java heap which is seen/shared by all threads.  You need to call `ThreadLocalRandom.current()` in each method that needs a `ThreadLocalRandom`.

Reply:
Okej, is it best to revert it back for now

Reply:
hey, what David means is instead of having the field, just use it every time.

E.g:

```java
public class FileUtils {

//not more the static field

    private File home;

    private FileUtils(final String homeDir, final String defaultDir) {
        this(homeDir, defaultDir, SystemInstance.get().getProperties());
    }

 public static File createTempDirectory(final String pathPrefix) throws IOException {
        for (int maxAttempts = 100; maxAttempts > 0; --maxAttempts) {

            final String path = pathPrefix + ThreadLocalRandom.current().nextLong();
            final File tmpDir = new File(path);

            if (!tmpDir.exists() && tmpDir.mkdirs()) {
                return tmpDir;
            }
        }
        throw new IOException("Cannot create temporary directory at: " + pathPrefix);
    }
```